app/models/miq_request_workflow.rb

```diff
@@ -1491,6 +1491,10 @@ def default_ci_to_hash_struct(ci)
     build_ci_hash_struct(ci, attributes)
   end
 
+  def ci_is_a_datacenter(ci)
+    ci.type&.end_with?("Datacenter")
```

Comment:
@agrare Since the Datacenter model is built on `EmsFolder` could we adding a property to the `MiqHashStruct` that is returned from `build_ci_hash_struct` based on checking `ci.kind_of?(Datacenter)` here

https://github.com/ManageIQ/manageiq/blob/3e7bb4fad773920c399ed99af69cf159b24e64da/app/models/miq_request_workflow.rb#L1011-L1013

Something like:
```ruby
def ems_folder_to_hash_struct(ci) 
  build_ci_hash_struct(ci, [:name, :type, :hidden]).tap do |hs|
    hs.send("datacenter?=", ci.kind_of?(Datacenter))
  end
end 
```

Reply:
:+1: that looks much better thanks